### PR TITLE
Os add sidepanel tests

### PR DESCRIPTION
**Pull request by slee1130**
### Purpose 🚀
_There were some testing missing in Sidepanel.spec.js so added some more scenarios to test.

### Notes ✏️
-Added sidepnal.group and sidepanel.trigger jest testing added
-Sidepanel.trigger testing could not use getByTestId since the test keeps failing if there is another way to test it please tell me I can fix it.
@oscarkwan 

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_
- [ ] MINOR (backward compatible) change to _these packages_
- [x] PATCH (bug fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/os-add_sidepanel_tests

### Screenshots 📸
_optional but highly recommended_

### References 🔗
_relevant Jira ticket / GitHub issues_

### CODE REVIEW

File: packages/SidePanel/tests/SidePanel.spec.js [hunk 1]
```diff
@@ -88,15 +87,96 @@ describe("SidePanel", () => {
...
+            <SidePanel.Header>With Header</SidePanel.Header>
+            <SidePanel.Overlay />
+          </SidePanel>
+          <SidePanel onClose={noop} isOpen>
+            <SidePanel.Header kind="primary">With Header</SidePanel.Header>
+            <SidePanel.Overlay />
+          </SidePanel>
+        </SidePanel.Group>
+      );
+      // data-pka-anchor
```
Comment (oscarkwan) [inline packages/SidePanel/tests/SidePanel.spec.js]: Please remove comment

File: packages/SidePanel/tests/SidePanel.spec.js [hunk 2]
```diff
@@ -88,15 +87,96 @@ describe("SidePanel", () => {
...
+      expect(getAllByTestId("heading")).toHaveLength(3);
```
Comment (oscarkwan) [inline packages/SidePanel/tests/SidePanel.spec.js]: @slee1130 Use the `sidepanel` to determine number of sidepanels and not the heading.

File: packages/SidePanel/tests/SidePanel.spec.js [hunk 3]
```diff
@@ -88,15 +87,96 @@ describe("SidePanel", () => {
...
+    });
+
+    it("throws the error when sidepanel has only one panel", () => {
```
Comment (oscarkwan) [inline packages/SidePanel/tests/SidePanel.spec.js]: "throws an error when there is only one side panel in a group"

File: packages/SidePanel/tests/SidePanel.spec.js [hunk 4]
```diff
@@ -88,15 +87,96 @@ describe("SidePanel", () => {
...
+            <SidePanel.Header>With Header</SidePanel.Header>
+            <SidePanel.Overlay />
+          </SidePanel>
+          <SidePanel onClose={noop} isOpen>
+            <SidePanel.Header kind="primary">With Header</SidePanel.Header>
+            <SidePanel.Overlay />
+          </SidePanel>
+        </SidePanel.Group>
+      );
+      fireEvent.click(getAllByTestId(/button/i)[0]);
```
Comment (oscarkwan) [inline packages/SidePanel/tests/SidePanel.spec.js]: we should be using `sidepanel-header-close`

File: packages/SidePanel/tests/SidePanel.spec.js (hunk 3 above)
Comment (slee1130) [reply packages/SidePanel/tests/SidePanel.spec.js]: done

File: packages/SidePanel/tests/SidePanel.spec.js (hunk 1 above)
Comment (slee1130) [reply packages/SidePanel/tests/SidePanel.spec.js]: done

File: packages/SidePanel/tests/SidePanel.spec.js [hunk 5]
```diff
@@ -88,15 +87,96 @@ describe("SidePanel", () => {
...
+    });
+
+    it("should tigger or render a button", () => {
+      const onClick = jest.fn();
+      const { getByRole } = render({
+        onClick,
+        children: <SidePanel.Trigger>Button</SidePanel.Trigger>,
+      });
+
+      expect(getByRole("button")).toBeInTheDocument();
```
Comment (oscarkwan) [inline packages/SidePanel/tests/SidePanel.spec.js]: You have the same expect in line 169. Which means this test is testing the same thing as above.

Therefore, it makes this test kind of redundant. I think just keeping "renders with default props" is fine.

File: packages/SidePanel/tests/SidePanel.spec.js (hunk 2 above)
Comment (slee1130) [reply packages/SidePanel/tests/SidePanel.spec.js]: done

File: packages/SidePanel/tests/SidePanel.spec.js (hunk 4 above)
Comment (slee1130) [reply packages/SidePanel/tests/SidePanel.spec.js]: done

File: packages/SidePanel/tests/SidePanel.spec.js (hunk 5 above)
Comment (slee1130) [reply packages/SidePanel/tests/SidePanel.spec.js]: done

**Pull request merged**